File: SowThistle.cpp
#include "CONSTANTS.h"
#include "SowThistle.h"

SowThistle::SowThistle(World &world, const Point &position) : Plant(world, position, SOWTHISTLE_SYMBOL, SOWTHISTLE_STRENGTH, SOWTHISTLE_COLOR) {}

void SowThistle::Action() {
  for (int i = 0; i < 3; i++) {
    Plant::Action();
  }
}

Organism* SowThistle::Clone(const Point &position) {
  return new SowThistle(this->world, position);
}

File: Java/SowThistle.java
public class SowThistle extends Plant {
  public SowThistle(World world, Point position) {
    super(world, position, Constants.SOW_THISTLE_SYMBOL, Constants.SOW_THISTLE_STRENGTH, Constants.SOW_THISTLE_COLOR, Constants.SOW_THISTLE_NAME);
  }

  @Override
  public Organism Clone(Point position) {
    return new SowThistle(this.world, position);
  }

  @Override
  public void Action() {
    for (int i = 0; i < 3; i++) {
      super.Action();
    }
  }
}
